bal_tools/subgraph.py

```diff
@@ -61,14 +61,48 @@ def get_subgraph_url(self, subgraph="core") -> str:
         returns:
         - https url of the subgraph
         """
+        if subgraph == "aura":
+            return AURA_SUBGRAPHS_BY_CHAIN.get(self.chain, None)
+
+        graph_api_key = os.getenv("GRAPH_API_KEY")
+        if not graph_api_key:
+            return self.get_subgraph_url_sdk(subgraph)
+
+        if subgraph == "core":
+            magic_word = "main:"
+        elif subgraph == "gauges":
+            magic_word = "gauge:"
+        elif subgraph == "blocks":
+            magic_word = "blocks:"
+
+        # get subgraph url from frontend config
+        chain_url_slug = 'gnosis-chain' if self.chain == 'gnosis' else self.chain
+        config_file = f"https://raw.githubusercontent.com/balancer/frontend-v2/develop/src/lib/config/{chain_url_slug}/index.ts"
+        found_magic_word = False
+        with urlopen(config_file) as f:
+            for line in f:
+                if found_magic_word or magic_word + " `" in str(line):
+                    # url is on this line
+                    r = re.search('`(.*)`', line.decode("utf-8"))
+                    try:
+                        url = r.group(1)
+                        if urlparse(url).scheme in ["http", "https"]:
+                            return url.replace('${keys.graph}', graph_api_key)
+                    except AttributeError:
+                        break
+                if magic_word in str(line):
+                    # url is on next line, return it on the next iteration
+                    found_magic_word = True
+        # loop again; config file might be of legacy type
+        return self.get_subgraph_url_legacy(subgraph, config_file)
```

Comment:
in the case where there is a graph api key but it cant find the url from the frontend config nor the legacy config, it should fallback to `get_subgraph_url_sdk`. right now, it will still fail to find the gnosis blocks graph if there is a graph api key

Reply:
thanks for keeping me sharp!

i wrote a unit test that finds all subgraphs (all chains, all subgraph types, with and without thegraph api key)

i should have started with writing that test really, then write the logic